### PR TITLE
Create SubmissionUnlockedBanner component

### PR DESCRIPTION
**Pull request by JasonLin0991**
[OY2-16188](https://qmacbis.atlassian.net/browse/OY2-16188)

## Summary
SubmissionUnlockedBanner to display banner for State and CMS users when submission has been unlocked.

#### Screenshots
CMS User Unlocked Banner
Screen Shot 2022-02-09 at 3 33 50 PM

State User Unlocked Banner
Screen Shot 2022-02-09 at 3 35 57 PM

#### Test cases covered

Added unit tests to make sure correct banner color displays for CMS and State users and passed in text is displayed on the banner.

### CODE REVIEW

File: services/app-web/src/components/SubmissionUnlockedBanner/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { SubmissionUnlockedBanner } from './SubmissionUnlockedBanner'
```
Comment (haworku) [inline services/app-web/src/components/SubmissionUnlockedBanner/index.ts:1]: Knowing that we have more of these alert banners coming up I would  like you to go ahead make the root folder and index.ts be something like `/Alert` or `/Banner` (similar to how we have `/Form`). Then this component  can live inside that folder.

File: services/app-web/src/components/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export const SubmissionUnlockedBanner = ({
+    userType,
+    unlockedBy,
+    unlockedOn,
+    reason,
+}: UnlockedProps): React.ReactElement => {
+    return (
+        <Alert role={'alert'} type={userType === 'CMS_USER' ? 'warning' : 'info'} heading={'Submission unlocked'} validation={true}>
+            <div className={styles.unlockedBanner}>
+                <p className={'usa-alert__text'}><b>Unlocked by: </b>{unlockedBy}</p>
```
Comment (haworku) [inline services/app-web/src/components/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx:27]:
```suggestion
                <p className={'usa-alert__text'}><b>Unlocked by:&nbsp;</b>{unlockedBy}</p>
```

File: services/app-web/src/components/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                <p className={'usa-alert__text'}><b>Unlocked on: </b>{dayjs(unlockedOn).format('MM/DD/YYYY hh:mma z')}</p>
```
Comment (haworku) [inline services/app-web/src/components/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx:28]:
```suggestion
                <p className={'usa-alert__text'}><b>Unlocked on:&nbsp;</b>{dayjs(unlockedOn).format('MM/DD/YYYY hh:mma z')}</p>
```

File: services/app-web/src/components/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                <p className={'usa-alert__text'}><b>Reason for unlock: </b>{reason}</p>
```
Comment (haworku) [inline services/app-web/src/components/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx:29]:
```suggestion
                <p className={'usa-alert__text'}><b>Reason for unlock:&nbsp;</b>{reason}</p>
```

**Review by haworku — APPROVED** [review]
Looks good! Will the PR actually adding this code to the application code come next? Let me know if you want to pair on that part.

**Review by haworku — APPROVED** [review]

**Comment by macrael** [comment]
@JasonLin0991 Yes I think that’s right. We need to thread the data through in order to put these in the code live.

File: services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    reason: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ut justo non nisl congue efficitur. Praesent porta condimentum imperdiet. Curabitur.',
+}
+
+export const SubmissionUnlockedBannerStateUser = Template.bind({})
+SubmissionUnlockedBannerStateUser.args = {
+    userType: 'STATE_USER',
+    unlockedBy: 'Loremipsum@email.com',
+    unlockedOn: new Date(),
+    reason: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ut justo non nisl congue efficitur. Praesent porta condimentum imperdiet. Curabitur.',
+}
```
Comment (macrael) [inline services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.stories.tsx:26]: Minor thing, but I reccomend turning on the “always have files end with newline” setting on your editor, I’m pretty sure the rest of us do.

File: services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import styles from './SubmissionUnlockedBanner.module.scss'
+import { Alert } from '@trussworks/react-uswds'
+import dayjs from 'dayjs'
+import timezone from 'dayjs/plugin/timezone.js'
+import advancedFormat from 'dayjs/plugin/advancedFormat.js'
+
+dayjs.extend(advancedFormat)
```
Comment (macrael) [inline services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx:8]: I don’t love seeing these global calls here. It looks like we are calling `dayjs.extend(isLeapYear)` in RateDetailSchmea as well. 

@haworku is this best practice? could we be requiring all our dayjs extensions somewhere more central? Or do you need to invoke dayjs.extend in every file where you use one of the extensions?

File: services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.test.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+                userType="STATE_USER"
+                unlockedBy="Loremipsum@email.com"
+                unlockedOn={testDate}
+                reason="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ut justo non nisl congue efficitur. Praesent porta condimentum imperdiet. Curabitur."
+            />
+        )
+        expect(screen.getByRole('alert')).toHaveClass('usa-alert--info')
+        expect(screen.getByText('Loremipsum@email.com')).toBeInTheDocument()
+        expect(screen.getByText(dayjs(testDate).format('MM/DD/YYYY hh:mma z'))).toBeInTheDocument()
+        expect(screen.getByText('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi ut justo non nisl congue efficitur. Praesent porta condimentum imperdiet. Curabitur.')).toBeInTheDocument()
```
Comment (macrael) [inline services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.test.tsx:40]: These tests are good, we don’t need much for components like this that have very little logic in them.

**Review by macrael — APPROVED** [review]
Great work!

File: services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx (hunk 6 above)
Comment (haworku) [reply services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx:8]: @macrael For sure agreed.  Much better to isolate in a lib or by applying it across the module once. We can make a `dateHelpers/dayjs.ts` placed at the root of app-web src. I think if we import into `index.tsx` and extend there it should work across the app (even when importing from the original `dayjs` but I'd have to check). If not we can make it a dayjs internal lib we reference to access this dependency.

Comment (JasonLin0991) [reply services/app-web/src/components/Banner/SubmissionUnlockedBanner/SubmissionUnlockedBanner.tsx]: @macrael @haworku  I put `dayjs` import and extensions in `dateHelpers/` directory to isolate and centralize the extensions. I also applied the change for `RateDetailSchmea`. Let me know what you think of this approach.

**Comment by macrael** [comment]
That looks great to me! @haworku does this look right to you?

**Review by haworku — APPROVED** [review]

**Pull request merged**